src/python/zquantum/core/interfaces/backend_test.py

```diff
@@ -132,6 +152,33 @@ def test_run_circuitset_and_measure(self, backend):
         else:
             assert backend.number_of_jobs_run == number_of_circuits
 
+    def test_run_circuitset_and_measure_n_samples(self, backend):
+        # Note: this test may fail with noisy devices
```

Comment:
I think I know how to tackle this so that we can keep both clean structure and minimal code overview. Would be happy to talk about it.